lib/updateVersion.js

```diff
@@ -0,0 +1,632 @@
+                Cartesian3.fromArray(rotation, 0, axis);
+                Quaternion.fromAxisAngle(axis, rotation[3], quat);
+                node.rotation = [quat.x, quat.y, quat.z, quat.w];
+            }
+            var instanceSkin = node.instanceSkin;
+            if (defined(instanceSkin)) {
+                node.skeletons = instanceSkin.skeletons;
+                node.skin = instanceSkin.skin;
+                node.meshes = instanceSkin.meshes;
+                delete node.instanceSkin;
+            }
+        }
+    }
+}
+
+function removeTechniquePasses(gltf) {
+    var techniques = gltf.techniques;
+    for (var techniqueId in techniques) {
+        if (techniques.hasOwnProperty(techniqueId)) {
+            var technique = techniques[techniqueId];
+            var passes = technique.passes;
+            if (defined(passes)) {
+                var passName = defaultValue(technique.pass, 'defaultPass');
+                if (passes.hasOwnProperty(passName)) {
+                    var pass = passes[passName];
+                    var instanceProgram = pass.instanceProgram;
+                    technique.attributes = defaultValue(technique.attributes, instanceProgram.attributes);
+                    technique.program = defaultValue(technique.program, instanceProgram.program);
+                    technique.uniforms = defaultValue(technique.uniforms, instanceProgram.uniforms);
+                    technique.states = defaultValue(technique.states, pass.states);
+                }
+                delete technique.passes;
+                delete technique.pass;
+            }
+        }
+    }
+}
+
+function glTF08to10(gltf) {
+    if (!defined(gltf.asset)) {
+        gltf.asset = {};
+    }
+    var asset = gltf.asset;
+    asset.version = '1.0';
+    // profile should be an object, not a string
+    if (!defined(asset.profile) || (typeof asset.profile === 'string')) {
+        asset.profile = {};
+    }
+    // version property should be in asset, not on the root element
+    if (defined(gltf.version)) {
+        delete gltf.version;
+    }
+    // material.instanceTechnique properties should be directly on the material
+    updateInstanceTechniques(gltf);
+    // primitive.primitive should be primitive.mode
+    setPrimitiveModes(gltf);
+    // node rotation should be quaternion, not axis-angle
+    // node.instanceSkin is deprecated
+    updateNodes(gltf);
+    // technique.pass and techniques.passes are deprecated
+    removeTechniquePasses(gltf);
+    // gltf.allExtensions -> extensionsUsed
+    if (defined(gltf.allExtensions)) {
+        gltf.extensionsUsed = gltf.allExtensions;
+        gltf.allExtensions = undefined;
+    }
+}
+
+function stripWebGLRevisionNumber(gltf) {
+    var asset = gltf.asset;
+    var profile = asset.profile;
+    if (defined(profile)) {
+        var version = profile.version;
+        if (defined(version)) {
+            profile.version = version[0] + '.' + version[2];
```

Comment:
This is correct as is, but a length check is a good idea.

| 0 | 1 | 2 | 3 | 4 |
|---|---|---|---|---|
| 1 | . | 0 | . | 3|

and we want `1.0`, so `version[0] + '.' + version[2]`.